modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/domain/DefaultDomainFactory.java

```diff
@@ -134,6 +134,12 @@ public void start(DomainWrapper domainWrapper) {
 
   @Override
   protected Domain doCreateArtifact(File domainLocation, Optional<Properties> deploymentProperties) throws IOException {
+    return createArtifact(domainLocation, deploymentProperties, null);
+  }
+
+  public Domain createArtifact(File domainLocation, Optional<Properties> deploymentProperties,
```

Comment:
what's the purpose of this split? where is this new method called?

Reply:
this should be used by tooling services